hw/dv/dpi/usbdpi/usb_monitor.c

```diff
@@ -172,59 +238,64 @@ void monitor_usb(void *mon_void, FILE *mon_file, int loglevel, int tick,
       uint32_t pkt_crc16, comp_crc16;
 
       if (compact && mon->byte == 2) {
-        fprintf(mon_file, "mon: %8d -- %8d: (%c) SOP, PID %s, EOP\n",
+        fprintf(mon->file, "mon: %8d -- %8d: (%c) SOP, PID %s, EOP\n",
                 mon->sopAt, tick, mon->driver == M_HOST ? 'H' : 'D',
                 pid_2data(mon->lastpid, mon->bytes[0], mon->bytes[1]));
       } else if (compact && mon->byte == 1) {
-        fprintf(mon_file, "mon: %8d -- %8d: (%c) SOP, PID %s %02x EOP\n",
+        fprintf(mon->file, "mon: %8d -- %8d: (%c) SOP, PID %s %02x EOP\n",
                 mon->sopAt, tick, mon->driver == M_HOST ? 'H' : 'D',
                 decode_pid[mon->lastpid & 0xf], mon->bytes[0]);
       } else {
         if (compact) {
-          fprintf(mon_file, "mon: %8d -- %8d: (%c) SOP, PID %s, EOP\n",
+          fprintf(mon->file, "mon: %8d -- %8d: (%c) SOP, PID %s, EOP\n",
                   mon->sopAt, tick, mon->driver == M_HOST ? 'H' : 'D',
                   decode_pid[mon->lastpid & 0xf]);
         }
-        fprintf(mon_file,
+        fprintf(mon->file,
                 "mon:     %s: ", mon->driver == M_HOST ? "h->d" : "d->h");
         comp_crc16 = CRC16(mon->bytes, mon->byte - 2);
-        pkt_crc16 = mon->bytes[mon->byte - 2] | mon->bytes[mon->byte - 1] << 8;
-        for (i = 0; i < mon->byte; i++) {
-          fprintf(mon_file, "%02x%s", mon->bytes[i],
-                  ((i & 0xf) == 0xf)       ? "\nmon:           "
-                  : ((i + 1) == mon->byte) ? ""
-                                           : ", ");
+        pkt_crc16 =
+            mon->bytes[mon->byte - 2] | (mon->bytes[mon->byte - 1] << 8);
+        // Display the data field as interleaved hexadecimal values and
+        // printable characters
+        for (i = 0; i < mon->byte - 2; i++) {
+          fprintf(mon->file, "%02x%s", mon->bytes[i],
+                  ((i & 0xf) == 0xf)           ? "\nmon:           "
+                  : ((i + 1) == mon->byte - 2) ? " "
+                                               : ", ");
           if ((mon->bytes[i] == 0x0d) || (mon->bytes[i] == 0x0a)) {
             mon->bytes[i] = '_';
-          }
-          if (mon->bytes[i] == 0) {
+          } else if (mon->bytes[i] == 0) {
             mon->bytes[i] = '?';
-          }
-          if (i >= (mon->byte - 2)) {
-            mon->bytes[i] = 0;
           } else if ((mon->bytes[i] < 32) || (mon->bytes[i] > 127)) {
             text = 0;
           }
         }
+        // Display the received CRC16 value
+        fprintf(mon->file, "(CRC16 %02x %02x", mon->bytes[mon->byte - 2],
+                mon->bytes[mon->byte - 1]);
         if (comp_crc16 == pkt_crc16) {
-          fprintf(mon_file, "%s CRCOK\n",
+          fprintf(mon->file, "%s OK)\n",
```

Comment:
Nit: Do you prefer 'OK' over 'CRCOK'?

Reply:
I've changed the log formatting to be more consistent with the other CRC (5-bit) reporting, as well as in a few other minor details to aid pairing up the log file with waveforms